src/tests/JIT/opt/Or/IntOr.cs

```diff
@@ -0,0 +1,112 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace CodeGenTests
+{
+    class IntOr
+    {
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void SideEffect()
+        {
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static bool Test_UInt32_UInt32_CastByte_Or(uint x, uint y)
+        {
+            // X64-NOT: movzx
+
+            // We expect 'or r8, r8'.
```

Comment:
In all of these, `r8` can be a bit confusing because it could literally mean the register `r8` or "any register of size 8".  `reg8` would be more obvious.